include/openssl/kdf.h

```diff
@@ -43,6 +43,9 @@ void EVP_KDF_CTX_reset(EVP_KDF_CTX *ctx);
 size_t EVP_KDF_CTX_get_kdf_size(EVP_KDF_CTX *ctx);
 int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t keylen,
                    const OSSL_PARAM params[]);
+int EVP_KDF_CTX_set_SKEY(EVP_KDF_CTX *ctx, EVP_SKEY *key);
+EVP_SKEY *EVP_KDF_derive_SKEY(EVP_KDF_CTX *ctx, const char *key_type,
+                              size_t keylen, const OSSL_PARAM params[]);
```

Comment:
Is there a reason to not pass the EVP_SKEY to the derive function directly?
This avoids needing `EVP_KDF_CTX_set_SKEY` at all and avoids the key_type parameter.

Reply:
Maybe if I worded it: "why isn't there an `EVP_SKEY *` argument for this call?"  It's got SKEY in the name so I'd expect it.  Having the SKEY passed to this call would avoid needing to call `EVP_KDF_CTX_set_SKEY` before.

Reply:
there may not be an input SKEY in some cases, and there are some cases when you need multiple ones in input. I am not against having an SKEY in input TBH, but it would have to be optional, and then *that* may be confusing?

Reply:
No more confusing than many of our APIs :grinning:

It was a question not a request for change and your response is reasonable.  Making it easiest for the most common case might be the way to go (I think this would be one SKEY but am not certain).

Reply:
Just to be clear, I am not against your request, I think it would make sense in many common cases.

Truth is until I can really test if this API is sufficient for handling Key derivation with pkcs11-provider, I am not sure.

I know for example that I have at least 2 key derivation operations where I can get multiple keys in output, and this API can't deal with that right now.